src/fits.cc

```diff
@@ -1128,6 +1128,47 @@ void Fits::closeFile() {
     fits_close_file(reinterpret_cast<fitsfile*>(fptr), &status);
 }
 
+
+PTR(daf::base::PropertyList) readMetadata(std::string const & fileName, int hdu, bool strip) {
+    fits::Fits fp(fileName, "r", fits::Fits::AUTO_CLOSE | fits::Fits::AUTO_CHECK);
+    fp.setHdu(hdu);
+    return readMetadata(fp, strip);
+}
+
+PTR(daf::base::PropertyList) readMetadata(fits::MemFileManager & manager, int hdu, bool strip) {
+    fits::Fits fp(manager, "r", fits::Fits::AUTO_CLOSE | fits::Fits::AUTO_CHECK);
+    fp.setHdu(hdu);
+    return readMetadata(fp, strip);
+}
+
+PTR(daf::base::PropertyList) readMetadata(fits::Fits & fitsfile, bool strip) {
+    PTR(daf::base::PropertyList) metadata(new lsst::daf::base::PropertyList);
```

Comment:
Is there's less repetitive way way to do this (and similar code below) with auto? Would this work?:
```
auto readMetadata = std::make_shared<daf::base::PropertyList>();
```